packages/dds/tree/src/feature-libraries/sequence-field/sequenceFieldEditor.ts

```diff
@@ -47,6 +64,80 @@ export const sequenceFieldEditor = {
                   detachedBy: revision,
                   detachIndex,
               }),
+    move(sourceIndex: number, count: number, destIndex: number): Changeset<never> {
+        if (count === 0 || sourceIndex === destIndex) {
+            // TODO: Should we allow creating a move which has no observable effect?
+            return [];
+        }
+
+        const moveOut: Mark<never> = {
+            type: "MoveOut",
+            id: brand(0),
+            count,
+        };
+
+        const moveIn: Mark<never> = {
+            type: "MoveIn",
+            id: brand(0),
+            count,
+        };
+
+        const factory = new MarkListFactory<never>();
+        if (sourceIndex < destIndex) {
```

Comment:
This doesn't take into account the possibility that `sourceIndex + count > destIndex`. Maybe add a TODO.

Reply:
Can you elaborate on what the problem is?

Reply:
Ah, sorry, I forgot about the fact that the destIndex is "after removal".
What I should have done is comment that the API prevents creating moves where the destination index lies between the detached nodes.
E.g., in a sequence [A, B, C], move(src: 0, count: 3, dst: 1).
This is a little odd but it is well-formed and meaningfully different from (say) move(src: 0, count: 3, dst: 0).
So we should either support such moves or provide a good reason why we don't want to allow them.